### ISSUE TITLE
Alerts for "elasticsearch" do not resolve "YYY-MM-DD-[*]" pattern in index name

### ISSUE THREAD

**Issue by kravchyuriy** [grafana/grafana#12731]
Hi everyone
Have some issue with configuring alerts for elasticsearch when I'm using this `YYYY-MM-DD-[*]` pattern  for "index name" field in "elasticsearch" data-source configuration.

Constantly getting this error : 
#### message:"Condition: Eval: false, Query Returned No Series (reduced to null/no value)"

This is the query which goes from grafana to elastic when I hit "Test Rule" button:
```
{
  "ignore_unavailable": true,
  "index": "YYYY-MM-DD[-metrics_staging",
  "max_concurrent_shard_requests": 256,
  "search_type": "query_then_fetch"
}
{
  "aggs": {
    "5": {
      "aggs": {
        "1": {
          "avg": {
            "field": "type_Memory.HeapMemoryUsage.used"
          }
        },
        "4": {
          "aggs": {
            "1": {
              "avg": {
                "field": "type_Memory.HeapMemoryUsage.used"
              }
            }
          },
          "date_histogram": {
            "field": "time",
            "interval": "1m",
            "min_doc_count": 0,
            "extended_bounds": {
              "min": "1532608950327",
              "max": "1532610630327"
            },
            "format": "epoch_millis"
          }
        }
      },
      "terms": {
        "field": "details.server.keyword",
        "size": 10,
        "order": {
          "1": "desc"
        },
        "min_doc_count": 100
      }
    }
  },
  "query": {
    "bool": {
      "filter": {
        "range": {
          "time": {
            "format": "epoch_millis",
            "gte": "1532608950327",
            "lte": "1532610630327"
          }
        }
      }
    }
  },
  "size": 0
}
```

As you can see from the query:
` "index": "YYYY-MM-DD[-metrics_staging"`
 - value for "index" key hasn't resolved itself, which leads to 'no data' state for alarm.
No such issues for quires, which update graphs:
```
{
  "search_type": "query_then_fetch",
  "ignore_unavailable": true,
  "index": [
    "2018-07-26-metrics_staging"
  ],
  "max_concurrent_shard_requests": 256
}
{
  "size": 0,
  "query": {
    "bool": {
      "filter": [
        {
        "range": {
            "time": {
              "gte": "1532608970364",
              "lte": "1532610770364",
              "format": "epoch_millis"
            }
          }
        },
        {
          "query_string": {
            "analyze_wildcard": true,
            "query": "*"
          }
        }
      ]
    }
  },
  "aggs": {
    "5": {
      "terms": {
        "field": "details.server.keyword",
        "size": 10,
        "order": {
          "1": "desc"
        },
        "min_doc_count": 100
      },
      "aggs": {
        "1": {
          "avg": {
            "field": "type_Memory.HeapMemoryUsage.used"
          }
        },
        "4": {
          "date_histogram": {
            "interval": "1m",
            "field": "time",
            "min_doc_count": 0,
            "extended_bounds": {
              "min": "1532608970364",
              "max": "1532610770364"
            },
            "format": "epoch_millis"
          },
          "aggs": {
            "1": {
              "avg": {
                "field": "type_Memory.HeapMemoryUsage.used"
              }
            }
          }
        }
      }
    }
  }
}
```
Also alerts work fine if using hard-coded index value (e.g. `2018-07-26-metrics_staging`) or wildcards (e.g. `YYYY-MM-DD-*`). 
But I need more specific value, which could equally substitute this one: `YYYY-MM-DD[-metrics_staging]` or maybe I'm missing something?

### What Grafana version are you using?
`v5.2.2`
### What datasource are you using?
`Elasticsearch`
### What OS are you running grafana on?
`Debian 9`
### What did you do?
Configured alarm for elastic-search data-source
### What was the expected result?
Alarm should trigger notification
### What happened instead?
got error:
`message:"Condition: Eval: false, Query Returned No Series (reduced to null/no value)"`

**Comment by marefr** [comment]
Only the pattern `[*]YYYY.MM.DD` is supported right now. Should be fixed. Thanks for reporting

**Comment by kravchyuriy** [comment]
Hi. Thanks for confirmation.

**Comment by marefr** [comment]
@kravchyuriy fixed in master now. You should be able to use a nightly build as soon as the master build completes. This is at the moment targeted to be included in Grafana v5.3 (no release data yet).
